tasks/office.py

```diff
@@ -66,34 +73,16 @@ def add_stored(self, object_key, object_data):
 
         if len(data_list) >= self.DATA_LIST_MAX:
             num_to_delete = random.randint(1, len(data_list))
-            del data_list[:num_to_delete]
+            # Convert a dict to a list so we can take a slice by insertion order fifo
+            self.local_store[object_key] = dict(list(data_list.items())[num_to_delete:])
 
         # Some creation endpoint auto-create multiple objects and return an array,
         # but each object in the array should still be considered individually here:
         if isinstance(object_data, list):
-            data_list.extend(object_data)
+            normalized = {value["id"]: value for value in object_data}
+            self.local_store[object_key] = {**normalized, **data_list}
```

Comment:
The use of `data_list` is starting to get confusing because it's not actually a list anymore. Would you be willing to change it to `data_dict` or something else that's a bit more informative?

Reply:
This is also unusual syntax and might be worth a comment. It is neat though. I think this mixin needs unit tests as well 🤔  (although you don't need to take point on adding those now - I also think we should standardize and combine this with the Prime storage mixin)

Reply:
Agreed, I added the comment to the top of the function but can talk about the combination here for which keys get overwritten.  Tests are a great idea and would give me more confidence as my Python brain is a little rusty.

Reply:
Tests could be a separate ticket imo! That's what Roci did, since folks had to get used to the testing environment. I'd be happy to pair on them too